### ISSUE TITLE
handlers not called after focus switch..

### ISSUE THREAD

**Issue by fantasticsid** [Chrisui/react-hotkeys#21]
Hi!

I have nested a hotkey element and an input element within another hotkey element:

```javascript
import React from 'react';
import {HotKeys} from 'react-hotkeys';

const Neat = React.createClass({
  onKeyDown (event) {
  },

  render () {
    var handlers = {
      'up': function(event) {
        console.log('up');
      },
      'down': function(event) {
        console.log('down');
      }
    };

    var inputHandlers = {
      'up': function(event) {
        console.log('inputHandler up');
      },
      'down': function(event) {
        console.log('inputHandler down');
      }
    };

    return (<div>
            <HotKeys handlers={inputHandlers}>
            <HotKeys handlers={handlers}>
            <div> hello </div>
            </HotKeys>
            <input ref='input' onKeyDown={this.onKeyDown}>
            </input>
            </HotKeys>
            </div>);
  }
});

module.exports = Neat;

```

Here's how I reproduced the issue:

1. click to focus on 'hello' div
2. use up and down key to trigger the handlers of the nested hotkey wrapper
3. click and focus on the input
4. now press up or down key, you'll find that one of the handlers of the outer layer is not called (and which one depends on the order of keys pressed in step 2..)

This is the minimal example to reproduce the problem I had.

The behavior depends on the order of keys pressed and whether you focus on a certain element. (stateful!) I wonder if it's something with mousetrap or the way we use it..

**Comment by fantasticsid** [comment]
It seems to be something with __lastChildSequence__, I'm not sure if I understand the check here: https://github.com/Chrisui/react-hotkeys/blob/master/lib/HotKeys.js#L110

So it's checking to see if current sequence is already handled by child, if child has processed the sequence, __lastChildSequence__ should already have been set to the sequence, thus parent should not process it.

But consider this case: 
1. child process sequence (e.g. 'a') 
2. __lastChildSequence__ of parent is set to 'a'
3. child loses focus, but focus is still within parent element
4. sequence 'a' is fired again

this is the case where child has not processed the sequence, while parent falsely believes it has. So it explains why the previous example did not work?

A fix would be to just reset __lastChildSequence__ of parent when a hotkey loses focus..

But I'm not sure if this will break other behaviors.

**Comment by Chrisui** [comment]
Thank you for the time taken to debug and solve this! I have merged your PR as you're totally correct that this state should've been cleaned up on blur.

I'm going to do a quick test and then I'll publish a release to npm.

Thanks again! :)

**Comment by fantasticsid** [comment]
Great! Thanks!

**Comment by fantasticsid** [comment]
Hey Chris,

Could you let me know when you do an update to npm. We'd rather pull it from npm than use our own copy of react-hotkeys.

Thank you!

**Comment by Chrisui** [comment]
Just published [v0.7.1](https://www.npmjs.com/package/react-hotkeys) with the fix for this.

CC @fantasticsid @pashasadri

Cheers!

**Comment by fantasticsid** [comment]
Awesome! Thanks Chris!
